### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by DarkDev-ai** [searx/searx#1901]
I have installed searx using this guide: https://github.com/asciimoo/searx/wiki/Installation

This worked but after messing up with the installation and reinstalling it only shows:
Internal Server Error

What could be wrong?
I'm using searx 16.0

**Comment by return42** [comment]
Why did you choose to follow the (old) wiki?

**Comment by DarkDev-ai** [comment]
Because the complete setup with docker is marked as "unfinished".

**Comment by dalf** [comment]
Despite some internal quirks, https://github.com/searx/searx-docker is working as intended, but the wiki page was not updated. Sorry for the misleading information.

**Comment by silvernode** [comment]
Same problem here, I updated with git stash and all that which has worked for over a year until this time I get internal server error. So I thought I should nuke my searx directory, and remove my searx-ve directory and once everything was back and running I got: internal server error just like OP. 

Yes I was using the (old) wiki, is there a (new) wiki by chance?

Edit: just to clarify I am not using docker but a traditional install on an Ubuntu 18.04 server with apache doing the frontend.

**Comment by dalf** [comment]
We need some log.

If you are using uwsgi:
* in ```/etc/uwsgi/apps-available/searx.ini```, remove ```disable-logging = true``` and restart uwsgi.
* see log in ```/var/log/uwsgi/app/searx.log```

OR start searx directly in debug mode:
```
export SEARX_DEBUG=1; python searx/webapp.py
```

BTW, which python version are using ?

**Comment by silvernode** [comment]
> We need some log.
> 
> If you are using uwsgi:
> 
>     * in `/etc/uwsgi/apps-available/searx.ini`, remove `disable-logging = true` and restart uwsgi.
> 
>     * see log in `/var/log/uwsgi/app/searx.log`
> 
> 
> OR start searx directly in debug mode:
> 
> ```
> export SEARX_DEBUG=1; python searx/webapp.py
> ```
> 
> BTW, which python version are using ?

Working on getting logs now, in the meantime I believe the version of Python is 2.7 which may be my problem if searx switch to python3 all of a sudden. I haven't really kept track of what version the project uses.

**Comment by silvernode** [comment]
This is the first and only error before uwsgi exits which happens immediately after starting the service,  

`ImportError: No module named site` 

This should not happen, I have downloaded and installed all the required modules and dependencies listed in the [installation page](https://asciimoo.github.io/searx/admin/installation.html).

### EDIT: I have fixed the issue by creating a new virtualenv with Python3:

For those of you reading this who already have an active install that stopped working like me, 
you will need to stop the uwsgi service and replace the `uwsgi-plugin-python` package with `uwsgi-plugin-python3`, and also install the package `python3-virtualenv` so you can create python3 virtual environments.  

For those of you using a fresh install, you will need to replace the Python packages that are listed on the [installation wiki page](https://asciimoo.github.io/searx/admin/installation.html), with the Python3 variants which should all be in your Linux distribution repositories. 

Below is what I did to migrate from an existing Python2/searx setup to a Python3 setup:

```
~:$ sudo /etc/init.d/uwsgi stop
~:$ sudo apt install python3-virtualenv python3-pip uwsgi-plugin-python3
~:$ sudo apt remove uwsgi-plugin-python
~:$ rm -R /usr/local/searx/searx-ve
~:$ virtualenv -p /usr/bin/python3 searx-ve
~:$ . ./searx-ve/bin/activate
~:$ ./manage.sh update_packages  
~:$ sudo /etc/init.d/uwsgi start
```
After doing this, you should not see the `Internal Server Error` message when you try and load your search instance.

If nobody is willing to update the docs for Python3, I can try doing it at some point when I have more time. It is probably most important now more than ever due to the fact that Python2 has been officially deprecated by its developers. 

I hope this helps somebody now and beyond. Until the docs get updated, I foresee there being many issues opened with this exact problem with fresh installs.

**Comment by return42** [comment]
> If nobody is willing to update the docs for Python3 ..

Thanks a lot, a new doc is on the way, see PR #1803 and the [preview](https://return42.github.io/searx/admin/installation.html).

This issue shows me that it is time to merge the #1803 .. hope I can finish my work these days ..

**Comment by silvernode** [comment]
> > If nobody is willing to update the docs for Python3 ..
> 
> Thanks a lot, a new doc is on the way, see PR #1803 and the [preview](https://return42.github.io/searx/admin/installation.html).
> 
> This issue shows me that it is time to merge the #1803 .. hope I can finish my work these days ..

This is exactly what we need and I am glad the time is here. That pull request should make things much better and easier manage. Thanks for sharing and thinks to everyone whole contributes to Searx.

**Comment by return42** [comment]
closed by PR #1803 / see docs at: https://asciimoo.github.io/searx/admin/installation.html
